### DISCUSSION TITLE
PTS100 T12 $15 Iron

### DISCUSSION THREAD

**Discussion by alextrical** [Ralim/IronOS#1690] [General]
Hi All

I was wondering if anyone else has come across these irons
![url.jpg](https://github.com/Ralim/IronOS/assets/35117191/44c11a83-7f00-4da0-b55e-0c749e8df09c)

 (I grabbed a couple as a backup to a pinecil V2) and was wondering how possible it would be to convert them to IronOS.
If you are interested, I'm happy to do the legwork to make a schematic, though I'm not familiar with the current programming language being used, but I'm also happy to learn

![20230525_224922.jpg](https://github.com/Ralim/IronOS/assets/35117191/a5dc13d8-f150-4fae-b963-bf3357343105)

![20230525_224929.jpg](https://github.com/Ralim/IronOS/assets/35117191/1fce4f8b-a1a3-4209-bb84-97a5d811d01e)

The main chip being used is a CHIPSEA CS32F030F6P6
It looks to be a ARM M0 (maybe a STM32 compatible?)
 https://datasheet.lcsc.com/lcsc/2110131830_CHIPSEA-CS32F030F6P6_C2905292.pdf

https://a.aliexpress.com/_mKdnsQ4

**Comment by alextrical** [comment]
I'm probably going to start tracing out the connections over the next week. It's seems to be a rather basic implementation.
Power is provided by a CH224K set to request 20V
A XL1509-3.3 is used to buck the voltage down to 3.3v
The MCU is a [CS32F030F6P6](http://www.wiseda-chip.com/en/showpro/?M1id117.html)

I'm hoping the ST-link V2 clone I have is able to connect to this over the 2 wire programming interface and dump a backup of the FW.

Before I invest time into this, does the MCU look like there is a chance of success to get running ironOS?

**Reply by Ralim** [reply]
Getting IronOS to run on that chip would be a really big stuggle due to the 4K of ram. The F0 chips are probably not something I would aim to support.

You _could_ if you had the means, swap it out for a pin compatible higher tier chip and it would then be quite doable.

Otherwise, to get 4K ram to work would need to very aggressively disable IronOS features to make that run at the moment. IronOS is really designed around ~20K of ram being available, so design descisions like OLED screen buffers etc are made with that in mind.
At the moment a "stock" build for the TS100 uses 12k of ram. It might be possible to trim things down to make it workable, but would be a bit of work.


The cpu core is fine, and you could definitely make the 32K of flash work otherwise.

I wouldn't have time in the near future to try and trim ram down that much.

**Reply by alextrical** [reply]
Thank you for getting back to me, and that is somewhat expected with a budget end of the chip range
In regards to replacing the chip the highest spec pin compatible is a [STM32C031F6P6](https://www.mouser.co.uk/ProductDetail/STMicroelectronics/STM32C031F6P6?qs=sGAEpiMZZMs0L%252B%252BydDbPCqCrcuy6EqN%252BOhpJ2npcWYkGWdZXX9V3MA%3D%3D)
with 32 KB Flash, 12 KB RAM. Not ideal, but the limit of the chip range.

I'm not expecting anyone to do the work for me, only to ask for some advice and guidance when I inevitably become stuck. Just hoping to help make the project better in some way or another for others too.

I'm wandering if its worth scratching the idea of using this iron. are there any good USB-C T12 or C245 handles out there that can run IronOS? If not I would be tempted to try and make an open source and probably 3D printed device.
I have a genuine Pinecil V2. I just don't like their tips that much

**Reply by Nomad990** [reply]
If it's v3 there's already a cool custom firmware for it just dont do it yet if its the v7 board i tried flashing it and it cuts info off the screen. 
https://github.com/spezifisch/T12PenSolder?tab=readme-ov-file

**Reply by Gobracket** [reply]
Forked and managed to fix the display for v7 board https://github.com/Gobracket/Gobracket-OS

Also simplified the display..
![MainDisplay](https://github.com/Ralim/IronOS/assets/129831736/8196da66-14a2-43df-a945-3063b3e1b554)

Need idea how to detect motion (for auto sleep - currently  based on timer).. Not sure if this model has built-in sensor / accelerometer, any thoughts anyone?

**Reply by alextrical** [reply]
That looks awesome. The unit I have does know when there is no movement and go to sleep after a few minutes of no movement.

**Reply by Gobracket** [reply]
Cool! Any idea how to enable it in the code? Thanks!

**Reply by alextrical** [reply]
I will take a look at my unit tomorrow and see what ist using to detect the movement, hopefully the board I have isn't too different from the version you have. 

My best guess is that motion is being detected with the white SMD chip beneath the MOSFET, that looks to be a vibration sensor similar to these 
https://ae01.alicdn.com/kf/Sf8e7368087db40a38b351d376d5debf1z.jpg_640x640q90.jpg

Assuming that chip will be connected to one of the GPIO on the MCU, the countdown timer could be reset every time it sees movement. 

I will take a look inside the unit tomorrow and see where the vibration switch is connected to the MCU, it looks like the schematic is ommiting that part.

Edit: the V3 board I have, the vibration sensor is connected to pin 6/ PA0. As for how to enable it in firmware, unfortunately I haven't played around with it on this device yet

**Reply by Gobracket** [reply]
`Thanks man! Code updated. Firmware now have the motion detector feature!

**Reply by jkoWAW** [reply]
> I will take a look at my unit tomorrow and see what ist using to detect the movement, hopefully the board I have isn't too different from the version you have.
> 
> My best guess is that motion is being detected with the white SMD chip beneath the MOSFET, that looks to be a vibration sensor similar to these https://ae01.alicdn.com/kf/Sf8e7368087db40a38b351d376d5debf1z.jpg_640x640q90.jpg
> 
> Assuming that chip will be connected to one of the GPIO on the MCU, the countdown timer could be reset every time it sees movement.
> 
> I will take a look inside the unit tomorrow and see where the vibration switch is connected to the MCU, it looks like the schematic is ommiting that part.
> 
> Edit: the V3 board I have, the vibration sensor is connected to pin 6/ PA0. As for how to enable it in firmware, unfortunately I haven't played around with it on this device yet

Hi, Alextrical,
sorry to take you time with old stories, but your post is actually only thing I was able to find on this soldering iron.
I have one and it has been a bit of challenge for me, as it only speaks chinese... Besides it
 is probably ok, it is warming up, looks well, so I keep hesitating before I will throw it away.
I even do not know the OEM producer, company TOOLTOP has been selling it on Aliexpress (->pic). That;s it.
I would appreciate very much a bit of advice how to change the firmware so that it is usable...
Thank you in Advance
Jacek
![t12 soldering iron](https://github.com/user-attachments/assets/a74f2060-5075-4005-9770-57272ec2f1d9)

**Reply by Gobracket** [reply]
Hi, you can read this: 
https://github.com/spezifisch/T12PenSolder

And this is my custom firmware that I forked from there:
https://github.com/Gobracket/Gobracket-OS

**Reply by jkoWAW** [reply]
:-) thanks!
